lib/ansible/modules/cloud/azure/azure_rm_resource.py

```diff
@@ -27,6 +27,7 @@
   url:
     description:
       - Azure RM Resource URL.
+      - Use * instead of subscription id and it will be replaced with default subscription.
```

Comment:
well, resource is known by ansible and comes from environment, so there may be different ways to get it:
- fill it manually
- do the lookup for environment variable
- anything else depending on the environment

But ansible modules just have it, so user won't need to worry about it.


I mean , for other tasks, you don't have to specify subscription id, so if you use URL in _resource module, it may need additional effort from the user, and this change is to avoid it.
Also users may not want to check in their subscription id into their playbooks.

Reply:
I could change "*" to, say, "{subscription}" or "{}", I don't know. but * seems a kind of obvious.

Reply:
whether it's real problem.... yes it may cause some difficulties, there's no single lookup for subscription id, so writing a playbook that works in any environment may be not possible / difficult. we encountered that while working on OpenShift for instance.

Reply:
second think on this problem. a normal resource url is like:
/subscriptions/<subs_id>/resourceGroups/xxxx/providers/providername/subtype/<resourcename>.

User will meet the problem to get subscription id when manually constructing this long string.  how about make url either be:
1. a url user copied from some one else
2. a dict contains resource group and resource name and provider_name, then module construct url for them.
3. name only. then construct  url from current resource group.

this follows behavior of exisiting azure modules handles referenced resources, like nic in vm module, and also avoid "*"